common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -281,10 +282,6 @@ public Object run() {
 
         logger.debug("java.nio.DirectByteBuffer.<init>(long, int): {}",
                 DIRECT_BUFFER_CONSTRUCTOR != null ? "available" : "unavailable");
-
-        if (direct != null) {
-            freeDirectBuffer(direct);
```

Comment:
removing this makes it easier to break cycle references and its not super important anyway